SystemServices/SystemServices.cpp

```diff
@@ -1657,7 +1655,7 @@ namespace WPEFramework {
         {
             bool status = false;
             try {
-                const device::SleepMode &mode = device::Host::getInstance().getPreferredSleepMode();
+                device::SleepMode mode = device::Host::getInstance().getPreferredSleepMode();
```

Comment:
The object should be mockable. For example, device::SleepMode can wrap a pointer to a mock. Another approach is std::reference_wrapper as polymorphism works for references but this changes the type which should probably be avoided.